Document issues with Live Preview in Codespaces (#1581)
opensafely/documentation: docs/getting-started/how-to/troubleshoot-common-codespaces-issues/index.md

If you wish to make changes that will apply whenever anyone opens your project, then consider using a [Workspace Settings](https://code.visualstudio.com/docs/getstarted/settings#_workspace-settings) file in your project.

If you wish your settings and extensions to be available anywhere you open VS Code (including in Codespaces), then consider using [Settings Sync](https://code.visualstudio.com/docs/editor/settings-sync).

## Previewing HTML files doesn't work

If you are using a codespace to author an HTML file, for example a report, you may wish to see a live preview of this and for this purpose many will turn to the [Live Preview](https://marketplace.visualstudio.com/items?itemName=ms-vscode.live-server) VS Code extension from Microsoft.
However, if you install and try to use this extension in a codespace, you might experiences errors such as _"This content is blocked. Contact the site owner to fix the issue."_ or the preview window appearing blank.

> This is a [known issue](https://github.com/microsoft/vscode-livepreview/issues/111) which is currently awaiting a fix from the extension's developers.

Reviewer: ```suggestion
This is a [currently known issue](https://github.com/microsoft/vscode-livepreview/issues/111) with the extension.
```

> ### Workaround

Reviewer: question: I guess this so, but does this work without doing the [dance described](https://github.com/microsoft/vscode-livepreview/issues/111#issuecomment-1095606237) in this post?

I know when I tried to get Live Preview working a month or two ago it was a bit of a hassle, and I'm not sure what steps I took to get it working.

When you first try to open a preview window via the context menu (right click), you may see a pop-up in the bottom right hand corner of your screeen advising you of a new "application running on port 3000":

> ![vs code popup notifying of application running on port 300](vscode-popup.png)

Reviewer: ```suggestion
![VS Code popup notifying of application running on port 3000.](vscode-popup.png)
```

Clicking "Open in Browser" should open a new tab where you will see a directory listing of the files in your workspace, and from which you can navigate to your desired html file:

> ![browser window showing project workspace directory listing](directory-listing.png)

Reviewer: ```suggestion
![Browser window showing project workspace directory listing.](directory-listing.png)
```

> You can also navigate to this window via the "ports" tab in VS Code:

Reviewer: ```suggestion
You can also navigate to this window via the "Ports" tab in VS Code:
```

> ![VS Code ports tab showing application running on port 3000](vscode-ports.png)

Reviewer: ```suggestion
![VS Code ports tab showing application running on port 3000.](vscode-ports.png)
```
